src/states/finitemps.jl

```diff
@@ -282,6 +282,35 @@ end
 
 Base.checkbounds(::Type{Bool}, ψ::FiniteMPS, i::Integer) = 1 <= i <= length(ψ)
 
+Base.@propagate_inbounds function Base.getindex(ψ::FiniteMPS, i::Int)
+    c = ψ.center
+
+    @boundscheck begin
+        (1 <= i <= length(ψ)) || throw(BoundsError(ψ, i))
```

Comment:
```suggestion
        checkbounds(ψ, i)
```
(this is defined just above here)